src/librustc_lint/types.rs

```diff
@@ -544,7 +544,7 @@ fn ty_is_known_nonnull<'tcx>(cx: &LateContext<'tcx>, ty: Ty<'tcx>, mode: CItemKi
                 return true;
             }
             for variant in &def.variants {
-                if let Some(field) = variant.transparent_newtype_field(tcx) {
+                if let Some(field) = cx.transparent_newtype_field(variant) {
                     if ty_is_known_nonnull(cx, field.ty(tcx, substs), mode) {
```

Comment:
Couldn't it just be a function in this module, like `ty_is_known_nonnull` is?

Reply:
Fixed!